addons/mail/static/src/messaging/entity/entity_core.js

```diff
@@ -0,0 +1,268 @@
+odoo.define('mail.messaging.entity.core', function (require) {
+'use strict';
+
+const { patchClassMethods, patchInstanceMethods } = require('mail.messaging.utils');
+
+const registry = {};
+
+//------------------------------------------------------------------------------
+// Private
+//------------------------------------------------------------------------------
+
+/**
+ * @private
+ * @param {string} entityName
+ * @returns {Object}
+ */
+function _getEntryFromEntityName(entityName) {
+    if (!registry[entityName]) {
+        registry[entityName] = {
+            dependencies: [],
+            factory: undefined,
+            name: entityName,
+            patches: [],
+        };
```

Comment:
I don't like that this getter is creating an entity. I think only registerNewEntity should do it, and the getter should crash otherwise.

But I think I understand that this might be needed if order of registration can't be guaranteed or that there are some kind of cycle at play.

Reply:
Yes, patch could be registered before entity itself, hence the getter creates an entry if it doesn't exist.